### ISSUE TITLE
Segfault when running armv7 Docker image from M1/Arm on macOS

### ISSUE THREAD

**Issue by cconstab** [dart-lang/sdk#54909]
Running dart pub get in an armv7 Docker image on macOS Arm results in a segfault. Happens on dart:stable and dart:beta

Reproduction steps (from an M1 Mac)

1. dart create -t server-shelf docker_test
2. cd docker_test
3. Replace `RUN dart pub get` in the Dockerfile with
```
RUN \
 cat /proc/cpuinfo ; \
 uname -a ;\
 dart pub get
```
4. docker build --progress=plain --platform=linux/arm/v7 .
5. Update Docker file to `FROM dart:beta AS build` and re run

<details>
  <summary>Dart Stable dump</summary>

```

cconstab@tarial docker_test % docker build  --progress=plain --platform=linux/arm/v7 .
#0 building with "desktop-linux" instance using docker driver

#1 [internal] load build definition from Dockerfile
#1 transferring dockerfile: 633B done
#1 DONE 0.0s

#2 [internal] load metadata for docker.io/library/dart:stable
#2 DONE 0.4s

#3 [internal] load .dockerignore
#3 transferring context: 127B done
#3 DONE 0.0s

#4 [build 1/6] FROM docker.io/library/dart:stable@sha256:7c08a296c364e2a70af9c4c457867f44c57afba6db3e78fb4865695fe344ef5c
#4 DONE 0.0s

#5 [internal] load build context
#5 transferring context: 290B done
#5 DONE 0.0s

#6 [build 2/6] WORKDIR /app
#6 CACHED

#7 [build 3/6] COPY pubspec.* ./
#7 CACHED

#8 [build 4/6] RUN  cat /proc/cpuinfo ;  uname -a ; dart pub get
#8 0.107 processor      : 0
#8 0.107 BogoMIPS       : 48.00
#8 0.107 Features       : fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm jscvt fcma lrcpc dcpop sha3 asimddp sha512 asimdfhm dit uscat ilrcpc flagm ssbs sb paca pacg dcpodp flagm2 frint
#8 0.107 CPU implementer        : 0x00
#8 0.107 CPU architecture: 8
#8 0.107 CPU variant    : 0x0
#8 0.107 CPU part       : 0x000
#8 0.107 CPU revision   : 0
#8 0.107
#8 0.107 processor      : 1
#8 0.107 BogoMIPS       : 48.00
#8 0.107 Features       : fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm jscvt fcma lrcpc dcpop sha3 asimddp sha512 asimdfhm dit uscat ilrcpc flagm ssbs sb paca pacg dcpodp flagm2 frint
#8 0.107 CPU implementer        : 0x00
#8 0.107 CPU architecture: 8
#8 0.107 CPU variant    : 0x0
#8 0.107 CPU part       : 0x000
#8 0.107 CPU revision   : 0
#8 0.107
#8 0.107 processor      : 2
#8 0.107 BogoMIPS       : 48.00
#8 0.107 Features       : fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm jscvt fcma lrcpc dcpop sha3 asimddp sha512 asimdfhm dit uscat ilrcpc flagm ssbs sb paca pacg dcpodp flagm2 frint
#8 0.107 CPU implementer        : 0x00
#8 0.107 CPU architecture: 8
#8 0.107 CPU variant    : 0x0
#8 0.107 CPU part       : 0x000
#8 0.107 CPU revision   : 0
#8 0.107
#8 0.107 processor      : 3
#8 0.107 BogoMIPS       : 48.00
#8 0.107 Features       : fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm jscvt fcma lrcpc dcpop sha3 asimddp sha512 asimdfhm dit uscat ilrcpc flagm ssbs sb paca pacg dcpodp flagm2 frint
#8 0.107 CPU implementer        : 0x00
#8 0.107 CPU architecture: 8
#8 0.107 CPU variant    : 0x0
#8 0.107 CPU part       : 0x000
#8 0.107 CPU revision   : 0
#8 0.107
#8 0.107 processor      : 4
#8 0.107 BogoMIPS       : 48.00
#8 0.107 Features       : fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm jscvt fcma lrcpc dcpop sha3 asimddp sha512 asimdfhm dit uscat ilrcpc flagm ssbs sb paca pacg dcpodp flagm2 frint
#8 0.107 CPU implementer        : 0x00
#8 0.107 CPU architecture: 8
#8 0.107 CPU variant    : 0x0
#8 0.107 CPU part       : 0x000
#8 0.107 CPU revision   : 0
#8 0.107
#8 0.107 processor      : 5
#8 0.107 BogoMIPS       : 48.00
#8 0.107 Features       : fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm jscvt fcma lrcpc dcpop sha3 asimddp sha512 asimdfhm dit uscat ilrcpc flagm ssbs sb paca pacg dcpodp flagm2 frint
#8 0.107 CPU implementer        : 0x00
#8 0.107 CPU architecture: 8
#8 0.107 CPU variant    : 0x0
#8 0.107 CPU part       : 0x000
#8 0.107 CPU revision   : 0
#8 0.107
#8 0.107 processor      : 6
#8 0.107 BogoMIPS       : 48.00
#8 0.107 Features       : fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm jscvt fcma lrcpc dcpop sha3 asimddp sha512 asimdfhm dit uscat ilrcpc flagm ssbs sb paca pacg dcpodp flagm2 frint
#8 0.107 CPU implementer        : 0x00
#8 0.107 CPU architecture: 8
#8 0.107 CPU variant    : 0x0
#8 0.107 CPU part       : 0x000
#8 0.107 CPU revision   : 0
#8 0.107
#8 0.107 processor      : 7
#8 0.107 BogoMIPS       : 48.00
#8 0.107 Features       : fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm jscvt fcma lrcpc dcpop sha3 asimddp sha512 asimdfhm dit uscat ilrcpc flagm ssbs sb paca pacg dcpodp flagm2 frint
#8 0.107 CPU implementer        : 0x00
#8 0.107 CPU architecture: 8
#8 0.107 CPU variant    : 0x0
#8 0.107 CPU part       : 0x000
#8 0.107 CPU revision   : 0
#8 0.107
#8 0.107 processor      : 8
#8 0.107 BogoMIPS       : 48.00
#8 0.107 Features       : fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm jscvt fcma lrcpc dcpop sha3 asimddp sha512 asimdfhm dit uscat ilrcpc flagm ssbs sb paca pacg dcpodp flagm2 frint
#8 0.107 CPU implementer        : 0x00
#8 0.107 CPU architecture: 8
#8 0.107 CPU variant    : 0x0
#8 0.107 CPU part       : 0x000
#8 0.107 CPU revision   : 0
#8 0.107
#8 0.107 processor      : 9
#8 0.107 BogoMIPS       : 48.00
#8 0.107 Features       : fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm jscvt fcma lrcpc dcpop sha3 asimddp sha512 asimdfhm dit uscat ilrcpc flagm ssbs sb paca pacg dcpodp flagm2 frint
#8 0.107 CPU implementer        : 0x00
#8 0.107 CPU architecture: 8
#8 0.107 CPU variant    : 0x0
#8 0.107 CPU part       : 0x000
#8 0.107 CPU revision   : 0
#8 0.107
#8 0.123 Linux buildkitsandbox 6.6.12-linuxkit #1 SMP Fri Jan 19 08:53:17 UTC 2024 armv7l GNU/Linux
#8 0.426
#8 0.426 ===== CRASH =====
#8 0.426 si_signo=Illegal instruction(4), si_code=2, si_addr=0x270c71c
#8 0.426 version=3.2.6 (stable) (Wed Jan 24 13:41:58 2024 +0000) on "linux_arm"
#8 0.426 pid=14, thread=18, isolate_group=vm-service(0x41100848), isolate=vm-service(0x41105940)
#8 0.426 os=linux, arch=arm, comp=no, sim=no
#8 0.426 isolate_instructions=21c9920, vm_instructions=21c9920
#8 0.426 fp=40fff84c, sp=40fff834, pc=270c71c
#8 0.427   pc 0x0270c71c fp 0x40fff84c Unknown symbol
#8 0.427   pc 0x0270c378 fp 0x40fff86c Unknown symbol
#8 0.427   pc 0x0270c154 fp 0x40fff88c Unknown symbol
#8 0.427   pc 0x0270bd38 fp 0x40fff8ac Unknown symbol
#8 0.427   pc 0x02700af0 fp 0x40fff8d0 Unknown symbol
#8 0.427   pc 0x02602a08 fp 0x40fff948 Unknown symbol
#8 0.427 -- End of DumpStackTrace
#8 0.428 qemu: uncaught target signal 6 (Aborted) - core dumped
#8 0.429 Aborted
#8 ERROR: process "/bin/sh -c cat /proc/cpuinfo ;  uname -a ; dart pub get" did not complete successfully: exit code: 134
------
 > [build 4/6] RUN  cat /proc/cpuinfo ;  uname -a ; dart pub get:
0.426 fp=40fff84c, sp=40fff834, pc=270c71c
0.427   pc 0x0270c71c fp 0x40fff84c Unknown symbol
0.427   pc 0x0270c378 fp 0x40fff86c Unknown symbol
0.427   pc 0x0270c154 fp 0x40fff88c Unknown symbol
0.427   pc 0x0270bd38 fp 0x40fff8ac Unknown symbol
0.427   pc 0x02700af0 fp 0x40fff8d0 Unknown symbol
0.427   pc 0x02602a08 fp 0x40fff948 Unknown symbol
0.427 -- End of DumpStackTrace
0.428 qemu: uncaught target signal 6 (Aborted) - core dumped
0.429 Aborted
------
Dockerfile:7
--------------------
   6 |     COPY pubspec.* ./
   7 | >>> RUN \
   8 | >>>  cat /proc/cpuinfo ; \
   9 | >>>  uname -a ;\
  10 | >>>  dart pub get
  11 |
--------------------
ERROR: failed to solve: process "/bin/sh -c cat /proc/cpuinfo ;  uname -a ; dart pub get" did not complete successfully: exit code: 134
cconstab@tarial docker_test %

```

</details>

<details>
  <summary>Dart Beta dump</summary>

```

cconstab@tarial docker_test % docker build  --progress=plain --platform=linux/arm/v7 .
#0 building with "desktop-linux" instance using docker driver

#1 [internal] load build definition from Dockerfile
#1 transferring dockerfile: 631B done
#1 DONE 0.0s

#2 [auth] library/dart:pull token for registry-1.docker.io
#2 DONE 0.0s

#3 [internal] load metadata for docker.io/library/dart:beta
#3 DONE 0.9s

#4 [internal] load .dockerignore
#4 transferring context: 127B done
#4 DONE 0.0s

#5 [build 1/6] FROM docker.io/library/dart:beta@sha256:3ebddc883daad8759c3bb8ec61119adee3ea97c5bec250a790f4a4b0d1deceac
#5 DONE 0.0s

#6 [internal] load build context
#6 transferring context: 290B done
#6 DONE 0.0s

#7 [build 2/6] WORKDIR /app
#7 CACHED

#8 [build 3/6] COPY pubspec.* ./
#8 CACHED

#9 [build 4/6] RUN  cat /proc/cpuinfo ;  uname -a ; dart pub get
#9 0.118 processor      : 0
#9 0.118 BogoMIPS       : 48.00
#9 0.118 Features       : fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm jscvt fcma lrcpc dcpop sha3 asimddp sha512 asimdfhm dit uscat ilrcpc flagm ssbs sb paca pacg dcpodp flagm2 frint
#9 0.118 CPU implementer        : 0x00
#9 0.118 CPU architecture: 8
#9 0.118 CPU variant    : 0x0
#9 0.118 CPU part       : 0x000
#9 0.118 CPU revision   : 0
#9 0.118
#9 0.118 processor      : 1
#9 0.118 BogoMIPS       : 48.00
#9 0.118 Features       : fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm jscvt fcma lrcpc dcpop sha3 asimddp sha512 asimdfhm dit uscat ilrcpc flagm ssbs sb paca pacg dcpodp flagm2 frint
#9 0.118 CPU implementer        : 0x00
#9 0.118 CPU architecture: 8
#9 0.118 CPU variant    : 0x0
#9 0.118 CPU part       : 0x000
#9 0.118 CPU revision   : 0
#9 0.118
#9 0.118 processor      : 2
#9 0.118 BogoMIPS       : 48.00
#9 0.118 Features       : fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm jscvt fcma lrcpc dcpop sha3 asimddp sha512 asimdfhm dit uscat ilrcpc flagm ssbs sb paca pacg dcpodp flagm2 frint
#9 0.118 CPU implementer        : 0x00
#9 0.118 CPU architecture: 8
#9 0.118 CPU variant    : 0x0
#9 0.118 CPU part       : 0x000
#9 0.118 CPU revision   : 0
#9 0.118
#9 0.118 processor      : 3
#9 0.118 BogoMIPS       : 48.00
#9 0.118 Features       : fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm jscvt fcma lrcpc dcpop sha3 asimddp sha512 asimdfhm dit uscat ilrcpc flagm ssbs sb paca pacg dcpodp flagm2 frint
#9 0.118 CPU implementer        : 0x00
#9 0.118 CPU architecture: 8
#9 0.118 CPU variant    : 0x0
#9 0.118 CPU part       : 0x000
#9 0.118 CPU revision   : 0
#9 0.118
#9 0.118 processor      : 4
#9 0.118 BogoMIPS       : 48.00
#9 0.118 Features       : fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm jscvt fcma lrcpc dcpop sha3 asimddp sha512 asimdfhm dit uscat ilrcpc flagm ssbs sb paca pacg dcpodp flagm2 frint
#9 0.118 CPU implementer        : 0x00
#9 0.118 CPU architecture: 8
#9 0.118 CPU variant    : 0x0
#9 0.118 CPU part       : 0x000
#9 0.118 CPU revision   : 0
#9 0.118
#9 0.118 processor      : 5
#9 0.118 BogoMIPS       : 48.00
#9 0.118 Features       : fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm jscvt fcma lrcpc dcpop sha3 asimddp sha512 asimdfhm dit uscat ilrcpc flagm ssbs sb paca pacg dcpodp flagm2 frint
#9 0.118 CPU implementer        : 0x00
#9 0.118 CPU architecture: 8
#9 0.118 CPU variant    : 0x0
#9 0.118 CPU part       : 0x000
#9 0.118 CPU revision   : 0
#9 0.118
#9 0.118 processor      : 6
#9 0.118 BogoMIPS       : 48.00
#9 0.118 Features       : fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm jscvt fcma lrcpc dcpop sha3 asimddp sha512 asimdfhm dit uscat ilrcpc flagm ssbs sb paca pacg dcpodp flagm2 frint
#9 0.118 CPU implementer        : 0x00
#9 0.118 CPU architecture: 8
#9 0.118 CPU variant    : 0x0
#9 0.118 CPU part       : 0x000
#9 0.118 CPU revision   : 0
#9 0.118
#9 0.118 processor      : 7
#9 0.118 BogoMIPS       : 48.00
#9 0.118 Features       : fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm jscvt fcma lrcpc dcpop sha3 asimddp sha512 asimdfhm dit uscat ilrcpc flagm ssbs sb paca pacg dcpodp flagm2 frint
#9 0.118 CPU implementer        : 0x00
#9 0.118 CPU architecture: 8
#9 0.118 CPU variant    : 0x0
#9 0.118 CPU part       : 0x000
#9 0.118 CPU revision   : 0
#9 0.118
#9 0.118 processor      : 8
#9 0.118 BogoMIPS       : 48.00
#9 0.118 Features       : fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm jscvt fcma lrcpc dcpop sha3 asimddp sha512 asimdfhm dit uscat ilrcpc flagm ssbs sb paca pacg dcpodp flagm2 frint
#9 0.118 CPU implementer        : 0x00
#9 0.118 CPU architecture: 8
#9 0.118 CPU variant    : 0x0
#9 0.118 CPU part       : 0x000
#9 0.118 CPU revision   : 0
#9 0.118
#9 0.118 processor      : 9
#9 0.118 BogoMIPS       : 48.00
#9 0.118 Features       : fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm jscvt fcma lrcpc dcpop sha3 asimddp sha512 asimdfhm dit uscat ilrcpc flagm ssbs sb paca pacg dcpodp flagm2 frint
#9 0.118 CPU implementer        : 0x00
#9 0.118 CPU architecture: 8
#9 0.118 CPU variant    : 0x0
#9 0.118 CPU part       : 0x000
#9 0.118 CPU revision   : 0
#9 0.118
#9 0.133 Linux buildkitsandbox 6.6.12-linuxkit #1 SMP Fri Jan 19 08:53:17 UTC 2024 armv7l GNU/Linux
#9 0.452
#9 0.452 ===== CRASH =====
#9 0.452 si_signo=Illegal instruction(4), si_code=ILL_ILLOPN(2), si_addr=0x278c45c
#9 0.453 version=3.3.0-279.3.beta (beta) (Wed Jan 31 15:24:56 2024 +0000) on "linux_arm"
#9 0.453 pid=14, thread=18, isolate_group=vm-service(0x41200848), isolate=vm-service(0x41205940)
#9 0.453 os=linux, arch=arm, comp=no, sim=no
#9 0.453 isolate_instructions=2248960, vm_instructions=2248960
#9 0.453 fp=410ff84c, sp=410ff834, pc=278c45c
#9 0.455   pc 0x0278c45c fp 0x410ff84c [Unoptimized] _IntegerImplementation@0150898.~/+0x54
#9 0.456   pc 0x0278c0b8 fp 0x410ff86c [Unoptimized] Timer._createTimer@4048458+0x1c8
#9 0.457   pc 0x0278be94 fp 0x410ff88c [Unoptimized] _RootZone@4048458.createTimer+0xac
#9 0.458   pc 0x0278ba78 fp 0x410ff8ac [Unoptimized] new Timer.+0x110
#9 0.459   pc 0x02780af0 fp 0x410ff8d0 [Unoptimized] main+0x3a0
#9 0.460   pc 0x02682a08 fp 0x410ff948 [Stub] InvokeDartCode+0xa8
#9 0.460 -- End of DumpStackTrace
#9 0.461 qemu: uncaught target signal 6 (Aborted) - core dumped
#9 0.462 Aborted
#9 ERROR: process "/bin/sh -c cat /proc/cpuinfo ;  uname -a ; dart pub get" did not complete successfully: exit code: 134
------
 > [build 4/6] RUN  cat /proc/cpuinfo ;  uname -a ; dart pub get:
0.453 fp=410ff84c, sp=410ff834, pc=278c45c
0.455   pc 0x0278c45c fp 0x410ff84c [Unoptimized] _IntegerImplementation@0150898.~/+0x54
0.456   pc 0x0278c0b8 fp 0x410ff86c [Unoptimized] Timer._createTimer@4048458+0x1c8
0.457   pc 0x0278be94 fp 0x410ff88c [Unoptimized] _RootZone@4048458.createTimer+0xac
0.458   pc 0x0278ba78 fp 0x410ff8ac [Unoptimized] new Timer.+0x110
0.459   pc 0x02780af0 fp 0x410ff8d0 [Unoptimized] main+0x3a0
0.460   pc 0x02682a08 fp 0x410ff948 [Stub] InvokeDartCode+0xa8
0.460 -- End of DumpStackTrace
0.461 qemu: uncaught target signal 6 (Aborted) - core dumped
0.462 Aborted
------
Dockerfile:7
--------------------
   6 |     COPY pubspec.* ./
   7 | >>> RUN \
   8 | >>>  cat /proc/cpuinfo ; \
   9 | >>>  uname -a ;\
  10 | >>>  dart pub get
  11 |
--------------------
ERROR: failed to solve: process "/bin/sh -c cat /proc/cpuinfo ;  uname -a ; dart pub get" did not complete successfully: exit code: 134
cconstab@tarial docker_test %

```

</details>

**Comment by mraleph** [comment]
Could you try to take [the `test` binary from this archive](https://github.com/dart-lang/sdk/files/14277135/test.zip) and include it into the `Dockerfile` and then paste the output?

```docker
COPY test /tmp/test
RUN /tmp/test 
```

This should give something like:

```
#7 [3/3] RUN /tmp/test
#7 0.422 hwcap fdfb0d7, contains HWCAP_ARM_IDIVA: true
#7 0.422 executing sdiv:
#7 0.422 4 / 4 = 1
#7 DONE 0.4s
```

The binary is compiled from the following source:

```cpp
#include <stdio.h>
#include <stdlib.h>
#include <sys/auxv.h>

int32_t sdiv(int32_t a, int32_t b) {
    int32_t c;
    asm("sdiv %0, %1, %2\n" : "=r"(c) : "r"(a), "r"(b));
    return c;
}

int main(int argc, char** argv) {
    unsigned long hwcap = getauxval(AT_HWCAP);
    fprintf(stderr, "hwcap %lx, contains HWCAP_ARM_IDIVA: %s\n", hwcap, (hwcap & HWCAP_ARM_IDIVA) ? "true" : "false");
    fprintf(stderr, "executing sdiv:\n");
    int32_t a = 16;
    int32_t b = 4;
    fprintf(stderr, "%d / %d = %d\n", sdiv(a, b));
    return 0;
}
```

Using the following command:

``` 
$ arm-linux-gnueabihf-gcc -static -mcpu=cortex-a7 -O3  -otest /tmp/test.c
```

**Comment by cconstab** [comment]
For sure NP, here are the results..

```
#10 0.101 hwcap 1fb0d7, contains HWCAP_ARM_IDIVA: true
#10 0.101 executing sdiv:
#10 0.101 4 / 4 = 1
```

<details>
  <summary>full docker output</summary>
  
```
cconstab@tarial docker_test % docker build --progress=plain --platform=linux/arm/v7 .
#0 building with "desktop-linux" instance using docker driver

#1 [internal] load build definition from Dockerfile
#1 transferring dockerfile: 661B done
#1 DONE 0.0s

#2 [internal] load metadata for docker.io/library/dart:beta
#2 ...

#3 [auth] library/dart:pull token for registry-1.docker.io
#3 DONE 0.0s

#2 [internal] load metadata for docker.io/library/dart:beta
#2 DONE 1.7s

#4 [internal] load .dockerignore
#4 transferring context: 127B done
#4 DONE 0.0s

#5 [build 1/7] FROM docker.io/library/dart:beta@sha256:3ebddc883daad8759c3bb8ec61119adee3ea97c5bec250a790f4a4b0d1deceac
#5 DONE 0.0s

#6 [internal] load build context
#6 transferring context: 431.07kB 0.0s done
#6 DONE 0.0s

#7 [build 2/7] WORKDIR /app
#7 CACHED

#8 [build 3/7] COPY pubspec.* ./
#8 CACHED

#9 [build 4/7] COPY test /tmp/test
#9 DONE 0.0s

#10 [build 5/7] RUN /tmp/test  cat /proc/cpuinfo ;  uname -a ; dart pub get
#10 0.101 hwcap 1fb0d7, contains HWCAP_ARM_IDIVA: true
#10 0.101 executing sdiv:
#10 0.101 4 / 4 = 1
#10 0.116 Linux buildkitsandbox 6.6.12-linuxkit #1 SMP Fri Jan 19 08:53:17 UTC 2024 armv7l GNU/Linux
#10 0.439 
#10 0.439 ===== CRASH =====
#10 0.439 si_signo=Illegal instruction(4), si_code=ILL_ILLOPN(2), si_addr=0x270c45c
#10 0.439 version=3.3.0-279.3.beta (beta) (Wed Jan 31 15:24:56 2024 +0000) on "linux_arm"
#10 0.439 pid=14, thread=18, isolate_group=vm-service(0x41200848), isolate=vm-service(0x41205940)
#10 0.439 os=linux, arch=arm, comp=no, sim=no
#10 0.439 isolate_instructions=2248960, vm_instructions=2248960
#10 0.439 fp=410ff84c, sp=410ff834, pc=270c45c
#10 0.441   pc 0x0270c45c fp 0x410ff84c [Unoptimized] _IntegerImplementation@0150898.~/+0x54
#10 0.442   pc 0x0270c0b8 fp 0x410ff86c [Unoptimized] Timer._createTimer@4048458+0x1c8
#10 0.443   pc 0x0270be94 fp 0x410ff88c [Unoptimized] _RootZone@4048458.createTimer+0xac
#10 0.444   pc 0x0270ba78 fp 0x410ff8ac [Unoptimized] new Timer.+0x110
#10 0.445   pc 0x02700af0 fp 0x410ff8d0 [Unoptimized] main+0x3a0
#10 0.446   pc 0x02682a08 fp 0x410ff948 [Stub] InvokeDartCode+0xa8
#10 0.446 -- End of DumpStackTrace
#10 0.447 qemu: uncaught target signal 6 (Aborted) - core dumped
#10 0.449 Aborted
#10 ERROR: process "/bin/sh -c /tmp/test  cat /proc/cpuinfo ;  uname -a ; dart pub get" did not complete successfully: exit code: 134
------
 > [build 5/7] RUN /tmp/test  cat /proc/cpuinfo ;  uname -a ; dart pub get:
0.439 fp=410ff84c, sp=410ff834, pc=270c45c
0.441   pc 0x0270c45c fp 0x410ff84c [Unoptimized] _IntegerImplementation@0150898.~/+0x54
0.442   pc 0x0270c0b8 fp 0x410ff86c [Unoptimized] Timer._createTimer@4048458+0x1c8
0.443   pc 0x0270be94 fp 0x410ff88c [Unoptimized] _RootZone@4048458.createTimer+0xac
0.444   pc 0x0270ba78 fp 0x410ff8ac [Unoptimized] new Timer.+0x110
0.445   pc 0x02700af0 fp 0x410ff8d0 [Unoptimized] main+0x3a0
0.446   pc 0x02682a08 fp 0x410ff948 [Stub] InvokeDartCode+0xa8
0.446 -- End of DumpStackTrace
0.447 qemu: uncaught target signal 6 (Aborted) - core dumped
0.449 Aborted
------
Dockerfile:8
--------------------
   7 |     COPY test /tmp/test
   8 | >>> RUN /tmp/test \
   9 | >>>  cat /proc/cpuinfo ; \
  10 | >>>  uname -a ;\
  11 | >>>  dart pub get
  12 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c /tmp/test  cat /proc/cpuinfo ;  uname -a ; dart pub get" did not complete successfully: exit code: 134
cconstab@tarial docker_test %
  
 ```
</details>

**Comment by mraleph** [comment]
Thanks @cconstab! This actually helped me to figure out what was going wrong with it

(The same problem can actually be reproduced on Linux - so it is not Docker on Mac specific at all). 

tldr is that our `SDIV` encoding was wrong. I think @rmacnak-google has a draft CL fixing it.
